modules/account-lib/src/coin/sol/transaction.ts

```diff
@@ -41,21 +44,145 @@ export class Transaction extends BaseTransaction {
   }
 
   /** @inheritdoc */
-  canSign(key: BaseKey): boolean {
-    throw new NotImplementedError('canSign not implemented');
+  canSign(): boolean {
+    return true;
   }
 
-  sign(keyPair: KeyPair | KeyPair[]): void {
-    throw new NotImplementedError('sign not implemented');
+  /**
+   * Signs transaction.
+   *
+   * @param {KeyPair} keyPair Signer keys.
+   */
+  async sign(keyPair: KeyPair[] | KeyPair): Promise<void> {
+    if (!this._solTransaction || !this._solTransaction.recentBlockhash) {
+      throw new SigningError('Nonce is required before signing');
+    }
+    if (!this._solTransaction || !this._solTransaction.feePayer) {
+      throw new SigningError('feePayer is required before signing');
+    }
+    const keyPairs = keyPair instanceof Array ? keyPair : [keyPair];
+    const signers: Signer[] = [];
+    for (const kp of keyPairs) {
+      const keys = kp.getKeys(true);
+      if (!keys.prv) {
+        throw new SigningError('Missing private key');
+      }
+      signers.push({ publicKey: new PublicKey(keys.pub), secretKey: keys.prv as Uint8Array });
+    }
+    try {
+      this._solTransaction.partialSign(...signers);
+    } catch (e) {
+      throw e;
+    }
   }
 
   /** @inheritdoc */
   toBroadcastFormat(): string {
-    throw new NotImplementedError('toBroadcastFormat not implemented');
+    if (!this._solTransaction) {
+      throw new ParseTransactionError('Empty transaction');
+    }
+    // The signatures can have null signatures (which means they are required but yet unsigned)
+    // In order to be able to serializer the txs, we have to change the requireAllSignatures based
+    // on if the TX is fully signed or not
+    const requireAllSignatures = requiresAllSignatures(this._solTransaction.signatures);
```

Comment:
nit: Please here add a comment in order to specify why when we got the broadcast format is on base64 and not base58 just only to clarify and inform